drivers/base/regmap/regmap-sdw-mbq.c

```diff
@@ -31,12 +32,40 @@ static int regmap_sdw_mbq_size(struct regmap_mbq_context *ctx, unsigned int reg)
 	return size;
 }
 
-static int regmap_sdw_mbq_write(void *context, unsigned int reg, unsigned int val)
+static int regmap_sdw_mbq_deferrable(struct regmap_mbq_context *ctx, unsigned int reg)
 {
-	struct regmap_mbq_context *ctx = context;
-	struct device *dev = ctx->dev;
-	struct sdw_slave *slave = dev_to_sdw_dev(dev);
-	int mbq_size = regmap_sdw_mbq_size(ctx, reg);
+	if (ctx->cfg.deferrable)
+		return ctx->cfg.deferrable(ctx->dev, reg);
+
+	return 0;
+}
+
+static int regmap_sdw_mbq_poll_busy(struct sdw_slave *slave, unsigned int reg,
+				    unsigned long timeout_us, unsigned long retry_us)
+{
+	struct device *dev = &slave->dev;
+	int val, ret;
+
+	dev_dbg(dev, "Deferring transaction for 0x%x\n", reg);
+
+	reg = SDW_SDCA_CTL(SDW_SDCA_CTL_FUNC(reg), 0, SDW_SDCA_FUNCTION_STATUS, 0);
+
+	ret = read_poll_timeout(sdw_read_no_pm, val,
+				val < 0 || !(val & SDW_SDCA_FUNCTION_BUSY),
+				timeout_us, retry_us, false, slave, reg);
```

Comment:
Also maybe I misunderstood the concept but would there be multiple regmap created/initialized for 8, 16 and 32 bits respectively?

In that case we probably need some sort of exclusion, no? Not sure if the bus msg_lock would work ?

Reply:
Again no my aim is to stop that happening, the current situation appears to have several drivers that are registering multiple register maps but that is just a war zone waiting to happen.